lib/GADS.pm

```diff
@@ -4801,6 +4805,9 @@ sub _process_edit
                 my $forward = (!$id && $layout->forward_record_after_create) || param('submit') eq 'submit-and-remain'
                     ? 'record/'.$record->current_id
                     : $layout->identifier.'/data';
+                $actions->{update_record} = $id ? $id: 0;
```

Comment:
I think this should maybe be called clear_saved_values or similar instead of update_record?